packages/sdk/src/chains/Substrate/__test__/getBasicFee.test.ts

```diff
@@ -1,48 +1,86 @@
+import { BigNumber } from 'ethers';
+import { FeeHandlerType } from '../../../types';
+import { XcmMultiAssetIdType, getBasicFee } from '../utils';
 
 const registry = new TypeRegistry();
 
-describe('getBasicFee', () => {
+describe('Substrate - getBasicFee', () => {
   it('should return the basic fee', async () => {
-    const result = new Option(registry, u128, '0x12345678');
+    const mockFee = '0x12345678';
+    const rawResult = new Option(registry, u128, mockFee);
+
     const api: ApiPromise = {
       query: {
         sygmaBasicFeeHandler: {
-          assetFees: jest.fn().mockResolvedValue(result),
+          assetFees: jest.fn().mockResolvedValue(rawResult),
         },
       },
     } as unknown as ApiPromise;
 
     const domainId = 1;
-    const xsmMultiAssetId = {};
+    const xsmMultiAssetId: XcmMultiAssetIdType = {
+      concrete: {
+        parents: 1,
+        interior: {
+          x3: [
+            {
+              parachain: 2004,
+            },
+            {
+              generalKey: [5, '0x12345'],
+            },
+            {
+              generalKey: [4, '0x1234'],
+            },
+          ],
+        },
+      },
+    };
 
     const feeRes = await getBasicFee(api, domainId, xsmMultiAssetId);
 
-    expect(feeRes).toBeInstanceOf(Option);
-    expect(feeRes.isSome).toBeTruthy();
-    expect(feeRes.unwrap()).toBeInstanceOf(u128);
+    expect(feeRes).toBeDefined();
+    expect(feeRes).toHaveProperty('fee');
+    expect(feeRes).toHaveProperty('type');
+    expect(feeRes.fee).toBeInstanceOf(BigNumber);
+    expect(feeRes.fee.eq(BigNumber.from(mockFee))).toBe(true);
+    expect(feeRes.type).toBe(FeeHandlerType.BASIC);
   });
 
-  it('should return none if the fee is not found', async () => {
-    const result = new Option(registry, u128, null);
+  it.skip('should throw and error if the fee is not found', () => {
+    const rawResult = new Option(registry, u128, null);
 
     const api: ApiPromise = {
       query: {
         sygmaBasicFeeHandler: {
-          assetFees: jest.fn().mockResolvedValue(result),
+          assetFees: jest.fn().mockResolvedValue(rawResult),
         },
       },
     } as unknown as ApiPromise;
     const domainId = 2; // some non-existent domain id;
-    const xsmMultiAssetId = {};
-
-    const feeRes = await getBasicFee(api, domainId, xsmMultiAssetId);
+    const xsmMultiAssetId: XcmMultiAssetIdType = {
+      concrete: {
+        parents: 1,
+        interior: {
+          x3: [
+            {
+              parachain: 2004,
+            },
+            {
+              generalKey: [5, '0x12345'],
+            },
+            {
+              generalKey: [4, '0x1234'],
+            },
+          ],
+        },
+      },
+    };
 
-    expect(feeRes).toBeInstanceOf(Option);
-    expect(feeRes.isNone).toBeTruthy();
+    const expectedError = new Error('Error retrieving fee');
+    expect(() => getBasicFee(api, domainId, xsmMultiAssetId)).toThrow(expectedError);
```

Comment:
minor suggestion here, just personal preference: you can async this test and do await expect instead of passing a function